models.js

```diff
@@ -4,19 +4,35 @@
  * https://bitbucket.org/javarosa/javarosa/src/tip/core/src/org/javarosa/xpath/expr/
  *
  */
-xpathmodels = {};
-xpathmodels.DEBUG_MODE = false;
 
-debuglog = function () {
-    if (xpathmodels.DEBUG_MODE) {
-        console.log(arguments);
-    }
-};
+if (!Function.prototype.bind) {
+    // PhantomJS doesn't support bind yet
+    Function.prototype.bind = function(oThis) {
+        if (typeof this !== 'function') {
+            // closest thing possible to the ECMAScript 5
+            // internal IsCallable function
+            throw new TypeError('Function.prototype.bind - what is trying to be bound is not callable');
+        }
+
+        var aArgs   = Array.prototype.slice.call(arguments, 1),
+            fToBind = this,
+            fNOP    = function() {},
+            fBound  = function() {
+                return fToBind.apply(this instanceof fNOP ? this : oThis,
+                                     aArgs.concat(Array.prototype.slice.call(arguments)));
+            };
 
-(function() {
-    var xpm = xpathmodels;
+        fNOP.prototype = this.prototype;
+        fBound.prototype = new fNOP();
 
-    var validateAxisName = xpm.validateAxisName = function(name) {
+        return fBound;
+    };
+}
+
+var XPathModels = function(hashtagConfig) {
```

Comment:
I know this name was my suggestion, but I think the convention is to only use function names starting with an uppercase letter *if they are to be instantiated using the `new` operator*, which this is not. Therefore I think we should rename this to something like `xPathModels` or `makeXPathModels`. Sorry for not thinking of this before.